modules/nf-core/fq/subsample/main.nf

```diff
@@ -0,0 +1,55 @@
+process FQ_SUBSAMPLE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::fq=0.9.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fq:0.9.1--h9ee0642_0':
+        'quay.io/biocontainers/fq:0.9.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(fastq)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"), emit: bam
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    /* args requires:
+        --probability <f64>: Probability read is kept, between 0 and 1. Mutually exclusive with record-count.
+        --record-count <u64>: Number of records to keep. Mutually exclusive with probability
+    */
+    def args = task.ext.args ?: ''
+    def prob_exists = args =~ /-p|--probability/
+    def nrec_exists = args =~ /-n|--record-count/
```

Comment:
Are these regexes tested? `-p` and `-n` can easily occur somewhere in the args so I would at least include word boundaries.